### ISSUE TITLE
Add Icons to Mark Type’s Select Box

### ISSUE THREAD

**Issue by kanitw** [vega/polestar#150]
Right now mark type's select box is HTML native select box.  It's much nicer to have some icons.

**Comment by ZeningQu** [comment]
It seems that the html <option> tags only allow text inside them. To add font awesome icons to the options I need some additional UI modules (see: http://angular-ui.github.io/). We are not using any UI modules right now, correct? 

I like the UI-bootstrap module (http://angular-ui.github.io/bootstrap/) because it covers a lot of things in the UI. But it's an overkill for issue #150. Do we want to use UI-bootstrap?

**Comment by kanitw** [comment]
You’re right. We can’t use html select box for this. 

For now, I prefer not to use bootstrap as bootstrap overrides too many default css, which makes developing other module difficult.   If you include bootstrap, basically all our UI components will suddenly becomes broken.  

For other dropdowns in the system we use Drop.js to implement. So it’s good to be consistent.  
For example see `fieldinfo.js` for example dropdown: 

The “include / optional / exclude” in  [Voyager](https://github.com/uwdata/voyager)’s `fieldlistitem.html` is a good example style that we can use with the mark type select box. 

![data_voyager](https://cloud.githubusercontent.com/assets/111269/7576686/7f511096-f7f5-11e4-9c15-b69fcff5db10.png)

**Comment by domoritz** [comment]
Icons are font awesome https://fortawesome.github.io/Font-Awesome/icons/

**Comment by kanitw** [comment]
@ZeningQu  

I was thinking about it again.  I think non-bootstrap UI components are generally fine as they don’t require all of bootstrap’s css.  Or if you want to use some bootstrap css, make sure to encapsulate your stylesheet within the module so it doesn’t override other parts. 

FYI, Dominik has start using `angular-ui-select` in another PR, so maybe you want to use that too.  It might be easier than what I suggested earlier.

**Comment by domoritz** [comment]
See https://github.com/angular-ui/ui-select

**Comment by ZeningQu** [comment]
@kanitw @domoritz Hey hey! Sorry it's been so long! I finished this using Drop.js (screenshot below):
![screen shot 2016-01-19 at 11 15 37 pm](https://cloud.githubusercontent.com/assets/822034/12442179/965de9ba-bf02-11e5-870f-bdaadb1092d9.png)
But this is only on my local machine. I tried to push to origin and got error 403:
![screen shot 2016-01-19 at 11 18 29 pm](https://cloud.githubusercontent.com/assets/822034/12442223/0ae7cf76-bf03-11e5-8d97-e8b74488de45.png)
